src/Extractor/HttpClient.php

```diff
@@ -313,12 +302,19 @@ private function possibleUnsupportedType($url)
      * Otherwise it will use the default one.
      *
      * @param string $url Absolute url
+     * @param array  $httpHeader Custom HTTP Headers from SiteConfig
      *
      * @return string
      */
-    private function getUserAgent($url)
+    private function getUserAgent($url, $httpHeader = array())
     {
         $ua = $this->config['ua_browser'];
+
+        if (array_key_exists('user-agent', $httpHeader) && !empty($httpHeader['user-agent'])) {
```

Comment:
You don't need the `array_key_exists()`, `empty()` already perform the `isset()`.